cogs/moderation.py

```diff
@@ -20,91 +129,216 @@ def __init__(self, bot: Bot) -> None:
+            )
+            embed.set_thumbnail(url=user.avatar_url_as(format="png", size=256))
+            await ctx.send(embed=embed)
+        except NotFound:
+            embed = discord.Embed(
+                title="No such ban",
+                description=textwrap.dedent(
+                    f"""
+                    There are no active bans on discord for {user.mention}.
+                    He isn't banned here.
+                    """
+                ),
+                color=discord.Color.red(),
+            )
+            await ctx.send(embed=embed)
 
     @command()
-    @bot_has_permissions(manage_messages=True)
     @has_permissions(manage_messages=True)
     async def clear(self, ctx: Context, amount: int) -> None:
         """Clear specified number of messages."""
+        await ctx.channel.purge(limit=amount + 1)
 
-        # TODO: Check if this condition is necessary
-        if amount is not None:
-            await ctx.channel.purge(limit=amount + 1)
-            # TODO: This message might be getting in the way,
-            # purpose of cleaning is to remove messages, not add more of them
-            await ctx.send("**Messages cleared** " + ctx.message.author.mention)
-            await asyncio.sleep(2.5)
-            await ctx.channel.purge(limit=1)
-        else:
-            await ctx.send("please specify the number of messages to clear")
+        embed = Embed(
+            description=textwrap.dedent(
+                f"""
+                **Messages Cleared!**
+                **Amount**: {amount}
+                """
+            ),
+            color=discord.Color.orange(),
+        )
+        message = await ctx.send(ctx.author.mention, embed=embed)
+        await asyncio.sleep(2.5)
+        await message.delete()
 
     @command()
-    @bot_has_permissions(manage_roles=True)
-    @has_permissions(manage_roles=True, manage_messages=True)
-    async def promote(self, ctx: Context, member: discord.Member, role: discord.Role) -> None:
+    @has_permissions(manage_roles=True)
+    async def promote(self, ctx: Context, member: Member, *, role: Role) -> None:
         """Promote member to role."""
-        # TODO: A custom check can handle this
         if role >= ctx.author.top_role:
-            embed = Embed(title="Error", description="Role needs to be below you in hierarchy.", color=Color.red(),)
+            embed = Embed(
+                title="Insufficient permissions",
+                description="You can give someone role which is higher in the role hierarchy than your top role.",
+                color=Color.red(),
+            )
             await ctx.send(embed=embed)
             return
         if role in member.roles:
-            embed = Embed(title="Error", description=f"{member.mention} already had role {role.mention}!", color=Color.red(),)
+            embed = Embed(title="Error", description=f"{member.mention} already have {role.mention} role!", color=Color.red())
```

Comment:
Small thingy, 
```suggestion
            embed = Embed(title="Error", description=f"{member.mention} already has the {role.mention} role!", color=Color.red())
```
Small thing, change the `already have` to `already has`